src/components/ha-bar-slider.ts

```diff
@@ -0,0 +1,410 @@
+import "hammerjs";
+import {
+  css,
+  CSSResultGroup,
+  html,
+  LitElement,
+  PropertyValues,
+  TemplateResult,
+} from "lit";
+import { customElement, property, query, state } from "lit/decorators";
+import { classMap } from "lit/directives/class-map";
+import { ifDefined } from "lit/directives/if-defined";
+import { styleMap } from "lit/directives/style-map";
+import { fireEvent } from "../common/dom/fire_event";
+
+declare global {
+  interface HASSDomEvents {
+    "slider-moved": { value?: number };
+  }
+}
+
+const A11Y_KEY_CODES = new Set([
+  "ArrowRight",
+  "ArrowUp",
+  "ArrowLeft",
+  "ArrowDown",
+  "PageUp",
+  "PageDown",
+  "Home",
+  "End",
+]);
+
+const getPercentageFromEvent = (e: HammerInput, vertical: boolean) => {
+  if (vertical) {
+    const y = e.center.y;
+    const offset = e.target.getBoundingClientRect().top;
+    const total = e.target.clientHeight;
+    return Math.max(Math.min(1, 1 - (y - offset) / total), 0);
+  }
+  const x = e.center.x;
+  const offset = e.target.getBoundingClientRect().left;
+  const total = e.target.clientWidth;
+  return Math.max(Math.min(1, (x - offset) / total), 0);
+};
+
+@customElement("ha-bar-slider")
+export class HaBarSlider extends LitElement {
+  @property({ type: Boolean }) public disabled = false;
+
+  @property({ attribute: "track-mode" })
+  public trackMode?: "indicator" | "start" | "end" = "start";
+
+  @property({ attribute: "orientation" })
+  public orientation?: "horizontal" | "vertical" = "horizontal";
+
+  @property({ attribute: false, type: Number, reflect: true })
```

Comment:
Why is this reflected and not an attribute?

Reply:
I remove it, it's not needed.